tez-runtime-library/src/main/java/org/apache/tez/dag/library/vertexmanager/FairShuffleVertexManager.java

```diff
@@ -192,18 +188,6 @@ public FairShuffleVertexManager(VertexManagerPluginContext context) {
     super(context);
   }
 
-  @Override
-  protected void onVertexStartedCheck() {
-    super.onVertexStartedCheck();
-    if (bipartiteSources > 1 &&
-        (mgrConfig.getFairRoutingType().fairParallelismEnabled())) {
-      // TODO TEZ-3500
-      throw new TezUncheckedException(
-          "Having more than one destination task process same partition(s) " +
-              "only works with one bipartite source.");
-    }
```

Comment:
I expect this assertion is not must because Tez users currently use FairShuffleVertexManager in specific cases on purpose.
I guess we need to add a new DataMovementType or new FairShuffleVertexManager for JOIN, but we don't immediately need it.